src/Adapter/Shop/Url/ProductPreview.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Shop\Url;
+
+use Link;
+use Tools;
+
+class ProductPreview
+{
+    /**
+     * @var Link
+     */
+    private $link;
+
+    /**
+     * @var int
+     */
+    private $employeeId;
+
+    public function __construct(
+        Link $link,
+        int $employeeId
+    ) {
+        $this->link = $link;
+        $this->employeeId = $employeeId;
+    }
+
+    public function getPreviewUrl(int $productId, bool $active): string
+    {
+        $preview_url = $this->link->getProductLink($productId);
```

Comment:
I'm a bit surprised however by the simplicity of the content here when I compare it to:
https://github.com/PrestaShop/PrestaShop/blob/96c92e602614db832229fc5d87c7becd3b965e01/controllers/admin/AdminProductsController.php#L2375-L2402

Reply:
I'm sure it works but it seems like your implementation handles less things and maybe the URL will look less like the one that will be used in FO. I guess the principle of this feature was to allow previewing the page of course, but maybe also what the URL would look like.

Seems like a few placeholders for URL rewriting are missing here (category, link_rerite, language, ...) Is it on purpose?

Reply:
It's may because the most part of the job is done by 
$preview_url = $this->link->getProductLink($productId);
on the first line of the method. (that already exists)
The preview only consist in adding additionnal parameters that it's done in the sprintf.
Did I respond to your question ?

Reply:
Ok I went into the details of `getProductLink` and indeed most of the parameters already handle their default value when you don't specify them and the default fallback is the same as what was explicitly specified in the previous code so it's ok.

Except for `$force_routes` parameter which is false by default, it should be specified based on `(bool) Configuration::get('PS_REWRITING_SETTINGS')` because with your current code I believe URLS will never use URL rewriting.

You can level the other parameters null or default values though.

Reply:
fixed